datashader/core.py

```diff
@@ -1131,10 +1148,173 @@ def raster(self,
             dims = [layer_dim]+dims
         return DataArray(data, coords=coords, dims=dims, attrs=attrs)
 
+    def volume(self, source, nan_value=None, agg='mean', interpolate='nearest'):
+        """Sample a raster dataset by canvas size and bounds.
+
+        Handles 3D xarray DataArrays.
+
+        Missing values (those having the value indicated by the
+        "nodata" attribute of the raster) are replaced with `NaN` if
+        floats, and 0 if int.
+
+        Parameters
+        ----------
+        source : xarray.DataArray or xr.Dataset
+            2D or 3D labelled array (if Dataset, the agg reduction must
+            define the data variable).
+        nan_value : int or float, optional
+            Optional nan_value which will be masked out when applying
+            the resampling.
```

Comment:
```suggestion
        nan_value : int or float, optional
            Optional nan_value that will be masked out when applying
            the resampling.
```

How does this interact with the "nodata" attribute of the raster? Does it override it? Should clarify that here.